include/RAJA/policy/cuda/policy.hpp

```diff
@@ -155,7 +155,7 @@ struct cuda_seq_syncthreads_exec
 ///////////////////////////////////////////////////////////////////////
 ///
 
-template <size_t BLOCK_SIZE, bool Async = false, bool maybe_atomic = false>
+template <bool Async = false, bool maybe_atomic = false>
```

Comment:
Having all the template parameters have default arguments means that the confusing `cuda_reduce<>` syntax is required. I personally don't like this, and I think it adds a burden to user code.

We could have an alias for this to keep user code cleaner.

Thoughts? @rchen20 @rhornung67 @ajkunen @trws

Reply:
@rchen20 I agree with @davidbeckingsale about removing unnecessary code rather than adding comments about it.  

Also, I don't think having all template parameters have defaults is a good idea since it hides potentially useful information and the cuda_reduce<> usage is confusing as @davidbeckingsale said.  We could remove the default value on the first parameter so synchronous/asynchronous execution is always explicit in user code. Then, I would use an enumeration for that arg not a bool; i.e., cuda_exec<true/false> is less clear than cuda_exec<async/sync>.

Reply:
@rhornung67 : I agree with you on the "sync/async" non-default cuda_exec argument, but think "maybe_atomic" should stay hidden/defaulted. 

One possible compromise would be to have your formulation: an enum for synchronous, but have type aliases

cuda_exec_synchronous and cuda_exec_asynchronous, I know from a readability perspective it would certainly help

Reply:
I think aliases is a good idea.

Reply:
The 2 comments have been removed, I mistakenly left those in there.

Regarding the cuda sync/async template parameter, I will make it a type alias with no enumeration, is that ok?

Also, "cuda_exec_async" is already defined here https://github.com/LLNL/RAJA/blob/fed74beecfa2c55a326ae3d736eb6b557fa5fd9c/include/RAJA/policy/cuda/policy.hpp#L213, and is used once here https://github.com/LLNL/RAJA/blob/fed74beecfa2c55a326ae3d736eb6b557fa5fd9c/test/unit/cuda/test-synchronize.cpp#L27. Do we want to rename it as cuda_x_sync/async, or something else?

Finally, to make sure I'm reading this correctly, asynchronous = false and synchronous = true?

Reply:
Correction, asynchronous = true and synchronous = false?

Reply:
We can keep the cuda_exec_async policy, and just add a cuda_exec_sync policy. You are correct w.r.t what true and false mean 😄

Reply:
Sorry, I should have been more clear. Having a type alias named "cuda_exec_async" would interfere with a policy already named as such. So, maybe we should reconsider our type alias naming to something like "using cuda_x_async = true;"? I was not going to create any new policy.

Reply:
I think the proper solution is like @rhornung67 suggested, ditch true/false for a sync/async enumeration. Then use aliases like:

`using cuda_exec_sync = cuda_exec<RAJA::cuda::sync>;`

Where `RAJA::cuda::sync` would be the enum value.

Reply:
@davidbeckingsale , might I suggest RAJA::sync (might conflict with a function)? That would allow easier switches between other backends which might want to use a synchronous or asynchronous variant of a kernel policy.